crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -319,11 +344,16 @@ impl SimpleFaucet {
                     // faucet fails or we give up before we get a definite response, we have a
                     // chance to retry later.
                     let mut wal = self.wal.lock().await;
-                    wal.reserve(uuid, coin_id, recipient, tx_data.clone())
-                        .map_err(FaucetError::internal)?;
+                    // lock all the coins in the id
+                    let mut coin_ids_clone = coin_ids.clone();
+                    while !coin_ids_clone.is_empty() {
+                        let coin_id = coin_ids_clone.pop().unwrap();
+                        wal.reserve(uuid, coin_id, recipient, tx_data.clone())
+                            .map_err(FaucetError::internal)?;
+                    }
```

Comment:
No need to clone and pop, just iterate over the `coin_ids`:

```suggestion
                    for coin_id in &coins_ids {
                        wal.reserve(uuid, *coin_id, recipient, tx_data.clone())
                            .map_err(FaucetError::internal)?;
                    }
```